lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2469,73 +2469,98 @@ namespace {
     }
 
     Decl *VisitNamespaceDecl(const clang::NamespaceDecl *decl) {
-      // If we have a name for this declaration, use it.
-      Optional<ImportedName> correctSwiftName;
-      auto importedName = importFullName(decl, correctSwiftName);
-      if (!importedName) return nullptr;
-
-      auto extensionDC =
-          Impl.importDeclContextOf(decl, importedName.getEffectiveContext());
-      if (!extensionDC)
-        return nullptr;
-
-      SourceLoc loc = Impl.importSourceLoc(decl->getBeginLoc());
       DeclContext *dc = nullptr;
       // If this is a top-level namespace, don't put it in the module we're
       // importing, put it in the "__ObjC" module that is implicitly imported.
       // This way, if we have multiple modules that all open the same namespace,
       // we won't import multiple enums with the same name in swift.
-      if (extensionDC->getContextKind() == DeclContextKind::FileUnit)
+      if (!decl->getParent()->isNamespace())
```

Comment:
This seems like a totally reasonable change to me. Just curious, did you notice a problem here, or is this just a cleanup?

Reply:
We're now retrieving `extensionDC` for each redeclaration individually (L2545) since `extensionDC` depends on the redecl's module, so we don't have `extensionDC` available here anymore.
We could still compute it for `decl`, checking the parent just takes less code.

Reply:
I see, thanks for explaining.